### PR TITLE
Add missing guard around new builtin.

### PR DESCRIPTION
**Pull request by nate-chandler**
Avoid a condfail.

### CODE REVIEW

**Review by tshortli — APPROVED** [review]

File: stdlib/public/core/UnsafeRawPointer.swift [hunk 1]
```diff
@@ -1316,7 +1316,9 @@ public struct UnsafeMutableRawPointer: _Pointer {
   public func storeBytes<T : _BitwiseCopyable>(
     of value: T, toByteOffset offset: Int = 0, as type: T.Type
   ) {
+#if $BuiltinStoreRaw
     Builtin.storeRaw(value, (self + offset)._rawValue)
+#endif
```
Comment (tshortli) [inline stdlib/public/core/UnsafeRawPointer.swift:1321]: Might want an `#else` branch with `fatalError()` here to minimize surprises for users of older compilers

**Comment by nate-chandler** [comment]
@swift-ci please smoke test

**Comment by nate-chandler** [comment]
@swift-ci please smoke test

**Comment by nate-chandler** [comment]
@swift-ci please smoke test linux platform

**Comment by nate-chandler** [comment]
@swift-ci Please test Linux

**Pull request merged**